### DISCUSSION TITLE
Grok-2 model ?

### DISCUSSION THREAD

**Discussion by mrtngrsbch** [danielmiessler/Fabric#1170] [General]
Hi,
Is there any project to integrate Grok ?

**Comment by scarnago** [comment]
Grok2 is a Openai like model, you can configure it using Openai plugin. But I get your point, to have a separate configuration, that way you don't need to choose Openai or Grok2, and we can get both configured.

**Reply by mrtngrsbch** [reply]
> Grok2 is a Openai like model, you can configure it using Openai plugin. But I get your point, to have a separate configuration, that way you don't need to choose Openai or Grok2, and we can get both configured.

Gracias Santiago,
Excuse my ignorance... does this mean that I can use the Grok2 keys in the OpenAI profile, directly and without any modification?
I would like to take advantage of the X Premium offer for Thanksgiving Day! 🤟🏼

**Reply by scarnago** [reply]
Yes, you are right, xAI announced $25 USD credit per month until the end of 2024. Announcement here: [https://x.ai/blog/api](https://x.ai/blog/api)

to use is on fabric app you just will replace chatGPT models from OpenAI by Grok model from xAI. Smart move from xAI, all their models are and will be OpenAI compatible.

API Endpoint address is: https://api.x.ai/v1
Just go to the: https://console.x.ai/ and create your API key, after that run `fabric --setup` in your terminal
 
Select AI Vendors -> [1] OpenAI

```
Enter your OpenAI API KEY (leave empty to skip): 
xai-xxxxxx 

Enter your OpenAI API BASE URL (leave empty for 'https://api.openai.com/v1' or type 'reset' to remove the value):
https://api.x.ai/v1
```

after that select Tools -> [10] Default AI Vendor and Model [required]
you will see a list with all your models from every vendor you have configured in my example case under OpenAI I have:

```
OpenAI
	[12]	grok-beta
	[13]	grok-vision-beta
```

I selected [12]

```
[Default]
Enter the index the name of your default model (leave empty for 'grok-beta' or type 'reset' to remove the value):
12

Enter model context length (leave empty for '122965' or type 'reset' to remove the value):
122965
```

Now you have Grok-2 available as default model

**Comment by scarnago** [comment]
After all that settings, you can test who's answering you.

```
santiago@deepcoolx:~$ echo "Who made you?" | fabric -p ai
- xAI's team of innovators and creators.
- Elon Musk, a visionary entrepreneur, leads xAI.
- A group dedicated to understanding the universe.
```

**Reply by mrtngrsbch** [reply]
many thanks Santiago !

**Reply by scarnago** [reply]
Hope you were able of made it work :)
